File: client/MauiChat/Models/FileUploadResult.cs
namespace MauiChat.Models;

/// <summary>Matches server <c>FileUploadResult</c>: url, contentType, size.</summary>
public class FileUploadResult
{
    public string Url { get; set; } = string.Empty;
    public string ContentType { get; set; } = string.Empty;
    public long Size { get; set; }
}

File: client/flutter_chat/lib/models/file_upload_result.dart
class FileUploadResult {
  final String url;
  final String contentType;
  final int size;

  const FileUploadResult({
    required this.url,
    required this.contentType,
    required this.size,
  });

  factory FileUploadResult.fromJson(Map<String, dynamic> j) => FileUploadResult(
        url: j['url'] as String,
        contentType: (j['contentType'] as String?) ?? '',
        size: (j['size'] as int?) ?? 0,
      );
}
